hypervisor/dm/vpci/vpci.c

```diff
@@ -340,6 +340,13 @@ static int32_t vpci_write_pt_dev_cfg(struct pci_vdev *vdev, uint32_t offset,
 	} else {
 		/* passthru to physical device */
 		pci_pdev_write_cfg(vdev->pdev->bdf, offset, bytes, val);
+
+		/* Write back BAR HPA when doing FLR */
+		if (vdev->has_flr && (vdev->pcie_capoff + PCIR_PCIE_DEVCTRL == offset) &&
+				((val & PCIM_PCIE_FLR) != 0U)) {
+			udelay(100000);	/* 100ms */
```

Comment:
Why 100ms? does spec say so?

Reply:
After an FLR has been initiated by writing a 1b to the Initiate Function Level Reset bit, the Function must complete the FLR within 100 ms. Will add comment for this.

Reply:
with core sharing based scheduler, we should not use udelay here. May add a new API to delay with schedule out